web/static/js/settings.js

```diff
@@ -264,13 +302,86 @@ $(function() {
+    else if ($(this).val() == "internetradio") { $("#internetradio_settings").show(); $("#addStreamButton").hide(); $("#streamNameDiv").hide(); }
     else if ($(this).val() == "fmradio") { $("#fmradio_settings").show(); $("#fmradio_warning").show(); }
     else if ($(this).val() == "plexamp") { $("#plexamp_settings").show(); }
 
   });
 
+  /* Hide some settings on the internet radio page when searching for stations */
+  $(document).on('shown.bs.tab', 'a[data-toggle="tab"]', function (e) {
+    e.target // newly activated tab
+    if (e.target.id == 'inetradio-search-name-tab' || e.target.id == 'inetradio-search-loc-tab') {
+      // Hide Stream Name field and Add Stream button
+      $("#addStreamButton").hide();
+      $("#streamNameDiv").hide();
+    }
+    else {
+      // Show Stream Name field and Add Stream button
+      $("#addStreamButton").show();
+      $("#streamNameDiv").show();
+    }
+  })
+
+  /* Search for inetradio stations */
+  $(document).on('click', '#inetradio-search-name-btn', function () {
+    console.log('Searching for station by name: ' + $("#inetradio-search-name-txt").val());
+    $.ajax({
+      type: "GET",
+      url: 'https://de1.api.radio-browser.info/json/stations/byname/' + $("#inetradio-search-name-txt").val(),
+      contentType: "application/json",
+      success: function(data) {
+        $("#inetradio-search_name_results").html("<h3>Search Results</h3>");
+        var details = '';
+        var numResults = 0;
+        $.each(data, function(index, value) {
+          ++numResults;
+          if (value.bitrate && value.codec) { details = '(' + value.bitrate + 'kbps ' + value.codec + ')'; }
+          $('#inetradio-search_name_results').append(
+            '<div style="position: relative; padding:6px;"><b><a href="' +
+            value.homepage +
+            '" target="_blank" title="Station Homepage">' +
+            value.name +
+            '</a></b> ' +
+            details +
+            '<a href="https://www.radio-browser.info/history/' +
+            value.stationuuid +
+            '" target="_blank" class="btn btn-info btn-sm float-right" role="button" style="position: absolute;right: 0;">More Info</a>' +
+            '<a href="#" class="btn btn-success btn-sm float-right addStation" data-stationname="' +
+            value.name.toString().replace('"', '\\"') +
+            '" data-stationurl="' +
+            value.url_resolved.toString().replace('"', '\\"') +
+            '" data-stationlogo="' +
+            value.favicon.toString().replace('"', '\\"') +
+            '" role="button" style="position: absolute;right: 90px;">Add Station</button></div>'
+          );
+        });
+        if (!numResults) { $('#inetradio-search_name_results').append("No stations found."); }
+      }
+    });
+  });
+
+  /* Add an inetradio station from search screen */
+  $(document).on('click', '.addStation', function () {
+    var name = $(this).data('stationname');
```

Comment:
This and all the associated hiding of the stream name setting means Internet Radio stations added by the search functionality don't allow custom names until after they are already added. Is this intentional? I'd think it would be nice to populate the name field with a selected radio station's name but allow editing before adding.

Reply:
I agree. This is going to need a little redesign to make it all work like that though.